matter/wifi/rs911x/rsi_if.c

```diff
@@ -318,27 +326,23 @@ static void wfx_rsi_do_join(void)
      * Right now it's done by hand - we need something better
      */
     wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTING;
-    while (((status = rsi_wlan_connect_async((int8_t *)&wfx_rsi.sec.ssid[0],
-                                             (rsi_security_mode_t)wfx_rsi.sec.security,
-                                             &wfx_rsi.sec.passkey[0],
-                                             wfx_rsi_join_cb))
-            != RSI_SUCCESS)
-           && (++wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)) {
+    while (((status = rsi_wlan_connect_async(
```

Comment:
This while check should be broken down to make it clearer what is actually beeing tested. With in one conditional statement, it is calling a function, setting a variable checkin the return of the function incremment a variable and checking it.

Reply:
Is it ok if i add comment to make it clear for understanding ?

Reply:
No, this should be changed.

Reply:
done

Reply:
How was this changed ? The while is not broken down like suggested.